comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,42 +10,125 @@ public enum Mark { None, O, X };
 
 	public class OxoBoard
 	{
-		// Constructor. Perform any necessary data initialisation here.
-		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+		Mark[,] boardSpaces = new Mark[3,3]; //Creates a new array using the Mark enumerable
+		public int fullSpaces; //creates an int value representing the filled spaces
+		List<Mark> occupiedSpacesX = new List<Mark>(); //Two seperate lists made to contain the used spaces for X and O
+		List<Mark> occupiedSpacesO = new List<Mark>();
+
+		public OxoBoard()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			boardSpaces.Initialize();
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			if (Mark.X.Equals(true))
+			{
+				occupiedSpacesX.Add(GetSquare(x, y)); //Adds the currently occupied space of X to the occupiedSpacesX list
```

Comment:
You are calling `GetSquare` from within `GetSquare`, which in this case causes infinite recursion. (That's why TravisCI fails with a `StackOverflowException`)